enhancements/baremetal/coreos-image-in-release.md

```diff
@@ -0,0 +1,258 @@
+the OpenShift release payload. When the metal3 Pod starts up, the disk image is
+copied from the container to a HostPath volume whence it is available to
+Ironic.
+
+The provisioning OS disk image is a separate CoreOS QCOW2 image. The URL for
+this is known by the installer. It points to the cloud by default and may be
+customised by the user to allow disconnected installs. The URL is stored in the
+Provisioning CR at install time and never updated automatically. The image
+itself is downloaded once and permanently cached on all of the master nodes.
+Never updating the image is tolerable because, upon booting, the CoreOS image
+will update itself to the version matching the cluster it is to join. It
+remains suboptimal because new Machines will take longer and longer (and more
+and more bandwidth) to join as the cluster ages. This issue exists on all
+platforms, and is the subject of a [long-standing enhancement
+proposal](https://github.com/openshift/enhancements/pull/201). Other issues
+specific to the baremetal platform are that boot times for bare metal servers
+can be very long (and therefore the reboot is costly), and that support for
+particular hardware may theoretically require a particular version of CoreOS.
+
+We are changing the deploy disk image to use CoreOS. This takes the form of
+both an ISO (for hosts that can use virtualmedia) and of a kernel + initrd +
+rootfs (for hosts that use PXE). When upgrading an existing disconnected
+cluster, we have no way to acquire these images without the user manually
+intervening to mirror them.
+
+Like the QCOW2 provisioning disk image, the URLs for these images are known by
+the installer, but they point to the cloud by default and would have to be
+customised by the user at install time to allow disconnected installs. IPA
+itself is delivered separately, as a container image as part of the OpenShift
+release payload.
+
+Following a similar approach to that currently used with the QCOW2 also
+effectively extends the limitation that we are not updating the provisioning OS
+image to include the deploy image as well (although we will continue to be able
+to update IPA itself).
+
+We wish to avoid these problems by obtaining an up-to-date CoreOS from the
+release payload.
+
+### Goals
+
+* Ensure that no matter which version of OpenShift a cluster was installed
+  with, we are able to deliver updates to IPA and the OS it runs on.
+* Stop maintaining and shipping the non-CoreOS, RHEL-based IPA PXE files.
+* Never break existing clusters, even if they are deployed in disconnected
+  environments.
+* Eliminate the extra reboot required to update CoreOS after initial
+  provisioning.
+
+### Non-Goals
+
+* Automatically switch pre-existing MachineSets to deploy with `coreos-install`
+  instead of via QCOW2 images.
+* Update the CoreOS QCOW2 image in the cluster with each OpenShift release.
+* Build CoreOS images for platforms other than baremetal.
```

Comment:
Maybe switch from "Build CoreOS images" to "Provide CoreOS images", since we're only talking about a way to provide existing built artifacts (then optionally embedding some config), not respinning any special/new images for this platform-specific usage?